mono/utils/mono-rand.c

```diff
@@ -35,6 +35,17 @@
 #include <sys/types.h>
 #include <sys/un.h>
 
+#if defined(__linux__) && defined(__GLIBC__)
+# include <linux/version.h>
+# if (LINUX_VERSION_CODE >= 0x31100) \
+     && (__GLIBC__ > 2 || __GLIBC__ == 2 && __GLIBC_MINOR__ >= 25)
+#  include <sys/random.h>
+#  ifndef HAVE_GETRANDOM
+#   define HAVE_GETRANDOM 1
```

Comment:
that should be in `configure.ac`

Reply:
something like that: https://hg.python.org/cpython/file/4243df51fe43/configure.ac#l5397

Reply:
I haven't used C in about 10 years so I'm afraid to mess with configure.ac. I just noticed that mono isn't using getrandom on Linux, when I tried to figure out how exactly KeePass2 was gathering entropy and generating random passwords. :)

Reply:
Would this work in `configure.ac` after line 2901? https://github.com/mono/mono/blob/master/configure.ac#L2902

I test compiled that short code snippet in Ubuntu 17.10 and it returned 0 as expected.

```
AC_MSG_CHECKING(for getrandom)
AC_CACHE_VAL(ac_cv_have_getrandom, [
AC_RUN_IFELSE([AC_LANG_SOURCE([[
#include <sys/random.h>

int main() {
    char buffer[1];
    const size_t buflen = sizeof(buffer);
    const int flags = 0;
    if (getrandom(buffer, buflen, flags) == buflen) {
        return 0;
    } else {
        return 1;
    }
}
]])],
[ac_cv_have_getrandom=yes],
[ac_cv_have_getrandom=no],
[ac_cv_have_getrandom=no])])
AC_MSG_RESULT($ac_cv_have_getrandom)
if test "x$ac_cv_have_getrandom" = "xyes"
then
  AC_DEFINE(HAVE_GETRANDOM, 1,
  [Define if getrandom() returns the number of bytes that were copied to the buffer.])
fi
```